src/components/Toast/Toast.tsx

```diff
@@ -0,0 +1,131 @@
+import * as React from 'react';
+import { FunctionComponent } from 'preact';
+import { useEffect, useRef, useState } from 'preact/hooks';
+import type { Toast as ToastPropType } from './types';
+import {
+  CheckCircleFilledIcon,
+  ClearCircleFilledIcon,
+} from '../../assets/icons';
+import { dismissToast } from '.';
```

Comment:
I believe you'd want to be careful importing from the index.ts file because it could lead to undefined imports, since the index file also references Toast.tsx and so you'd kind of be going in a loop. To avoid future headaches I also just import explicitly from the right file.